File: cpp/sum-of-multiples/sum_of_multiples.cpp
#include "sum_of_multiples.h"
#include <vector>
#include <map>

namespace sum_of_multiples {
    int to(std::vector<int> nums, int limit) {
        std::map<int, bool> multiples {};
        for (int num : nums) {
            for (int i = num; i < limit; i += num) multiples[i] = true;
        }
        int sum{0};
        for (auto &[key, _] : multiples) sum += key;
        return sum;
    }
}  // namespace sum_of_multiples

File: elixir/sum-of-multiples/lib/sum_of_multiples.ex
defmodule SumOfMultiples do
  @doc """
  Adds up all numbers from 1 to a given end number that are multiples of the factors provided.
  """
  @spec to(non_neg_integer, [non_neg_integer]) :: non_neg_integer
  def to(limit, factors) do
    factors
    |> Enum.map(&multiples(limit, &1))
    |> Enum.concat
    |> Enum.uniq
    |> Enum.sum
  end

  defp multiples(_, 0), do: []
  defp multiples(limit, factor), do: factor..(limit-1)//factor
end
